src/cli/cli.cpp

```diff
@@ -3365,35 +3365,6 @@ template <> otError Interpreter::Process<Cmd("ipmaddr")>(Arg aArgs[])
         SuccessOrExit(error = aArgs[1].ParseAsIp6Address(address));
         error = otIp6UnsubscribeMulticastAddress(GetInstancePtr(), &address);
     }
-    /**
-     * @cli ipmaddr promiscuous
-     * @code
-     * ipmaddr promiscuous
-     * Disabled
-     * Done
-     * @endcode
-     * @par api_copy
-     * #otIp6IsMulticastPromiscuousEnabled
-     */
-    else if (aArgs[0] == "promiscuous")
```

Comment:
We should remove the use of the `ipmaddr promiscuous` command across the codebase. It is currently found in:
*   `src/cli/README.md`
*   `tests/scripts/expect/cli-ipmaddr.exp`
*   `tools/otci/otci.py` and `tools/otci/tests/test_otci.py`

I used `git grep "ipmaddr promiscuous"` to search for it:
```bash
abtink ~/github/openthread (main) $ git grep "ipmaddr promiscuous"
src/cli/README.md:### ipmaddr promiscuous
src/cli/README.md:> ipmaddr promiscuous
src/cli/README.md:### ipmaddr promiscuous enable
src/cli/README.md:> ipmaddr promiscuous enable
src/cli/README.md:### ipmaddr promiscuous disable
src/cli/README.md:> ipmaddr promiscuous disable
src/cli/cli.cpp:     * @cli ipmaddr promiscuous
src/cli/cli.cpp:     * ipmaddr promiscuous
src/cli/cli.cpp:         * @cli ipmaddr promiscuous (enable,disable)
src/cli/cli.cpp:         * ipmaddr promiscuous enable
src/cli/cli.cpp:         * ipmaddr promiscuous disable
src/cli/cli.cpp:         * @cparam ipmaddr promiscuous @ca{enable|disable}
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous enable\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous disable\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous\n"
tools/otci/otci/otci.py:        return self.__parse_Enabled_or_Disabled(self.execute_command("ipmaddr promiscuous"))
tools/otci/otci/otci.py:        self.execute_command('ipmaddr promiscuous enable')
tools/otci/otci/otci.py:        self.execute_command('ipmaddr promiscuous disable')
tools/otci/tests/test_otci.py:        logging.info("ipmaddr promiscuous: %r", leader.get_ipmaddr_promiscuous())
```